src/RobustAdaptiveMetropolis.jl

```diff
@@ -0,0 +1,206 @@
+module RobustAdaptiveMetropolis
+
+using Random, LogDensityProblems, LinearAlgebra, AbstractMCMC
+using DocStringExtensions: FIELDS
+
+using AdvancedMH: AdvancedMH
+
+export RAM
+
+# TODO: Should we generalise this arbitrary symmetric proposals?
+# TODO: Implement checking of eigenvalues to avoid degenerate covariance estimates, as in the paper.
+"""
+    RAM
+
+Robust Adaptive Metropolis-Hastings (RAM).
+
+This is a simple implementation of the RAM algorithm described in [^VIH12].
+
+# Fields
+
+$(FIELDS)
+
+# Examples
+
+The following demonstrates how to implement a simple Gaussian model and sample from it using the RAM algorithm.
+
+```jldoctest
+julia> using AdvancedMH, Random, Distributions, MCMCChains, LogDensityProblems, LinearAlgebra
+
+julia> # Define a Gaussian with zero mean and some covariance.
+       struct Gaussian{A}
+           Σ::A
+       end
+
+julia> # Implement the LogDensityProblems interface.
+       LogDensityProblems.dimension(model::Gaussian) = size(model.Σ, 1)
+
+julia> function LogDensityProblems.logdensity(model::Gaussian, x)
+           d = LogDensityProblems.dimension(model)
+           return logpdf(MvNormal(zeros(d),model.Σ), x)
+       end
+
+julia> LogDensityProblems.capabilities(::Gaussian) = LogDensityProblems.LogDensityOrder{0}()
+
+julia> # Construct the model. We'll use a correlation of 0.5.
+       model = Gaussian([1.0 0.5; 0.5 1.0]);
+
+julia> # Number of samples we want in the resulting chain.
+       num_samples = 10_000;
+
+julia> # Number of warmup steps, i.e. the number of steps to adapt the covariance of the proposal.
+       # Note that these are not included in the resulting chain, as `discard_initial=num_warmup`
+       # by default in the `sample` call. To include them, pass `discard_initial=0` to `sample`.
+       num_warmup = 10_000;
+
+julia> # Set the seed so get some consistency.
+       Random.seed!(1234);
+
+julia> # Sample!
+       chain = sample(model, RAM(), 10_000; chain_type=Chains, num_warmup=10_000, progress=false, initial_params=zeros(2));
+
+julia> norm(cov(Array(chain)) - [1.0 0.5; 0.5 1.0]) < 0.2
+true
```

Comment:
Could we add more tests? Probably outside of the doctest. Currently we only test one of the constructor calls, and not e.g. passing a custom `S`. Also, I think the eigenvalue bound checking is probably untested, given the default values have a short-circuit path.